tests/cli/bootsys/test_mgmt_power.py

```diff
@@ -177,3 +183,230 @@ def test_mgmt_ncns_skip_prompt_power_off(self):
+        self.assertEqual(cm.records[1].message, f'Successfully set next boot device to disk (Boot0001) for ncn-s001')
+
+    def test_set_next_boot_device_to_disk_ssh_fail(self):
+        """Test that the function handles SSH connection failures and continues"""
+        mock_ssh_client = MagicMock()
+        mock_stdout = MagicMock()
+        mock_stderr = MagicMock()
+        mock_stdout.channel.recv_exit_status.return_value = 0
+        mock_stdout.read.return_value = b"Boot0001* UEFI OS\nBoot0002* Other OS"
+
+        # Set up the SSH connection side effects to simulate different scenarios
+        mock_ssh_client.connect.side_effect = [None, SSHException('ssh failed'), None]
+        mock_ssh_client.exec_command.return_value = (None, mock_stdout, mock_stderr)
+
+        with patch('sat.cli.bootsys.mgmt_power.LOGGER') as mock_logger:
+            # Test handling of SSH failure and continuation for all NCNs
+            ncns = ['ncn-w001', 'ncn-w002', 'ncn-w003']
+            set_next_boot_device_to_disk(mock_ssh_client, ncns)
+
+            # Check log messages and method calls
+            expected_connect_calls = [call('ncn-w001'), call('ncn-w002'), call('ncn-w003')]
+            mock_ssh_client.connect.assert_has_calls(expected_connect_calls)
+
+            expected_exec_calls = [
+                call('efibootmgr'),
+                call('efibootmgr -n 0001'),
+                call('efibootmgr'),
+                call('efibootmgr -n 0001')
+            ]
+            mock_ssh_client.exec_command.assert_has_calls(expected_exec_calls, any_order=True)
+
+            # Verify the log messages for successful and failed connections
+            mock_logger.info.assert_any_call('Successfully set next boot device to disk (Boot0001) for ncn-w001')
+            mock_logger.warning.assert_called_with('Unable to connect to node ncn-w002: ssh failed')
+            mock_logger.info.assert_any_call('Successfully set next boot device to disk (Boot0001) for ncn-w003')
+
+    def test_set_next_boot_device_to_disk_command_fail(self):
+        """Test that the function handles command execution failures and continues"""
+        # Mock successful connection for all NCNs
+        self.mock_ssh_client.connect.side_effect = [None, None, None]
+
+        # Mock command execution failure for the first NCN and success for the rest
+        mock_stdout_1 = MagicMock()
+        mock_stdout_2 = MagicMock()
+        mock_stderr = MagicMock()
+        mock_stdout_1.channel.recv_exit_status.return_value = 1
+        mock_stdout_1.read.return_value = b""
+        mock_stdout_2.channel.recv_exit_status.return_value = 0
+        mock_stdout_2.read.return_value = b"Boot0001* UEFI OS\nBoot0002* Other OS"
+
+        self.mock_ssh_client.exec_command.side_effect = [
+            (None, mock_stdout_1, mock_stderr),
+            (None, mock_stdout_2, mock_stderr),
+            (None, mock_stdout_2, mock_stderr)
```

Comment:
Set the exec_command side_effect such that it first simulates a failure of `efibootmgr`, then a successful `efibootmgr` and a successful `efibootmgr -n` for each of the other two NCNs.

```suggestion
            (None, mock_stdout_3, mock_stderr),
            (None, mock_stdout_2, mock_stderr),
            (None, mock_stdout_3, mock_stderr),
```